concordium-consensus/src/Concordium/KonsensusV1/Consensus.hs

```diff
@@ -31,4 +26,67 @@ class MonadMulticast m where
 newtype BakerContext = BakerContext
     { _bakerIdentity :: BakerIdentity
     }
-makeClassy ''BakerContext
\ No newline at end of file
+
+makeClassy ''BakerContext
+
+-- |A Monad for timer related actions.
+class MonadTimeout m where
+    -- |Reset the timeout from the supplied 'Duration'.
+    resetTimer :: Duration -> m ()
+
+-- |Produce a block and multicast it onto the network.
+makeBlock :: MonadState (SkovData (MPV m)) m => m ()
+makeBlock = return ()
+
+-- |Make a block if the consensus runner is leader for the
+-- current round.
+-- TODO: call 'makeBlock' if we're leader for the current round.
+makeBlockIfLeader :: MonadState (SkovData (MPV m)) m => m ()
+makeBlockIfLeader = return ()
+
+-- |Advance to the provided 'Round'.
+--
+-- This function does the following:
+-- * Update the current 'RoundStatus'.
+-- * Persist the new 'RoundStatus'.
+-- * If the consensus runner is leader in the new
+--   round then make the new block.
+advanceRound ::
+    ( MonadReader r m,
+      HasBakerContext r,
+      MonadTimeout m,
+      LowLevel.MonadTreeStateStore m,
+      MonadState (SkovData (MPV m)) m
+    ) =>
+    -- |The 'Round' to progress to.
+    Round ->
+    -- |If we are advancing from a round that timed out
+    -- then this will be @Left 'TimeoutCertificate, 'QuorumCertificate')@
+    -- The 'TimeoutCertificate' is from the round we're
+    -- advancing from and the associated 'QuorumCertificate' verifies it.
+    --
+    -- Otherwise if we're progressing via a 'QuorumCertificate' then @Right QuorumCertificate@
+    -- should be the QC we're advancing round via.
+    Either (TimeoutCertificate, QuorumCertificate) QuorumCertificate ->
+    m ()
+advanceRound newRound timedOut = do
```

Comment:
`timedOut` also has a value if we haven't timed out, so maybe this could be better named. Maybe `newCertificates`?

Reply:
Yes. I agree that the current name is..... let's say "not ideal" :) .. 

I'll update it.